pkg/docker/config/config.go

```diff
@@ -85,27 +90,36 @@ func GetAuthentication(sys *types.SystemContext, registry string) (string, strin
 	}
 
 	dockerLegacyPath := filepath.Join(homedir.Get(), dockerLegacyHomePath)
-	var paths []string
-	pathToAuth, err := getPathToAuth(sys)
+	paths := []authPath{}
+	pathToAuth, lf, err := getPathToAuth(sys)
 	if err == nil {
-		paths = append(paths, pathToAuth)
+		paths = append(paths, authPath{path: pathToAuth, legacyFormat: lf})
 	} else {
 		// Error means that the path set for XDG_RUNTIME_DIR does not exist
 		// but we don't want to completely fail in the case that the user is pulling a public image
 		// Logging the error as a warning instead and moving on to pulling the image
 		logrus.Warnf("%v: Trying to pull image in the event that it is a public image.", err)
 	}
-	paths = append(paths, filepath.Join(homedir.Get(), dockerHomePath), dockerLegacyPath)
+	homeDirDockerCFG := authPath{
+		path:         dockerLegacyPath,
+		legacyFormat: true,
+	}
+	homeDirConfigJSON := authPath{
+		path:         filepath.Join(homedir.Get(), dockerHomePath),
+		legacyFormat: false,
+	}
+	paths = append(paths, homeDirConfigJSON, homeDirDockerCFG)
 
 	for _, path := range paths {
-		legacyFormat := path == dockerLegacyPath
-		username, password, err := findAuthentication(registry, path, legacyFormat)
+		// in addition to the home dir, openshift may mount .dockercfg files (via secret mount)
+		// in locations other than the home dir
+		username, password, err := findAuthentication(registry, path.path, path.legacyFormat)
```

Comment:
Recording the motivation is definitely useful, but it’s unclear to me what the reader of the comment is supposed to do about it _at this point_. It would be a better fit in the `getPathToAuth` that deals with the `LegacyFormatAuthFilePath` option, in `types.go` declaring the option, or maybe only in a commit message.

Reply:
I moved it to types.go

IMO it will be much harder to find these details if they are only in a commit